earth_enterprise/src/fusion/dbmanifest/dbmanifest_unittest.cpp

```diff
@@ -119,24 +146,102 @@ class DbManifestTest : public testing::Test {
+      const PoiEntry& poi_entry = params.poi_entries[idx];
+      std::string actual_poi_file_path;
+
+      if (!params.prefix_loc.empty()) {
+        actual_poi_file_path = khComposePath(params.prefix_loc, poi_entry.poi_file);
+      }
+      else {
+        actual_poi_file_path = poi_entry.poi_file;
+      }
+      // add element to header xml
+      AddElement(poi_file_paths_elm, "poi_file_path", poi_entry.poi_file);
+
+      // Create the mock poi file
+      khDomDeleteGuard poi_file_dom(TransferOwnership(CreateEmptyDocument("POISearchFile")));
+      khxml::DOMElement* poi_file_root_elm = poi_file_dom->getDocumentElement();
+      khxml::DOMElement* search_table_values_elm = poi_file_dom->createElement(ToXMLStr("SearchTableValues"));
+      poi_file_root_elm->appendChild(search_table_values_elm);
+
+      for (size_t jdx = 0; jdx < poi_entry.poi_data_files.size(); ++jdx) {
+        const std::string& poi_data_file_path = poi_entry.poi_data_files[jdx];
+        std::string actual_poi_data_file_path;
+
+        if (!params.prefix_loc.empty()) {
+          actual_poi_data_file_path = khComposePath(params.prefix_loc, poi_data_file_path);
+        }
+        else {
+          actual_poi_data_file_path = poi_data_file_path;
+        }
+        AddElement(search_table_values_elm, "SearchDataFile", poi_data_file_path);
+
+        // write out the poi data file (empty file for these tests)
+        khEnsureParentDir(actual_poi_data_file_path);
+        khMakeEmptyFile(actual_poi_data_file_path);
+      }
+
+      // write out the poi file
+      khEnsureParentDir(actual_poi_file_path);
+      WriteDocument(poi_file_dom, actual_poi_file_path);
+    }
+
     // write out the header file
     khEnsureParentDir(actual_db_header_path);
     WriteDocument(dom, actual_db_header_path);
 
-    // Create the fake poi file
-    khDomDeleteGuard poi_file_dom(TransferOwnership(CreateEmptyDocument("POISearchFile")));
-    khxml::DOMElement* poi_file_root_elm = poi_file_dom->getDocumentElement();
-    khxml::DOMElement* search_table_values_elm = poi_file_dom->createElement(ToXMLStr("SearchTableValues"));
-    poi_file_root_elm->appendChild(search_table_values_elm);
-    AddElement(search_table_values_elm, "SearchDataFile", poi_data_file_path_);
+    // create a dummy empty index file
+    khEnsureParentDir(actual_toc_path);
+    khMakeEmptyFile(actual_toc_path);
+  }
+
+  bool IsEquivalent(std::vector<ManifestEntry>* manifest1,
+                    std::vector<ManifestEntry>* manifest2) {
 
-    // write out the poi file
-    khEnsureParentDir(actual_poi_file_path);
-    WriteDocument(poi_file_dom, actual_poi_file_path);
+    // sort no matter what.  Makes things easier to figure out
+    // what is missing when we dump the manifests
+    std::sort(manifest1->begin(), manifest1->end(), LessThan);
+    std::sort(manifest2->begin(), manifest2->end(), LessThan);
+    if (manifest1->size() != manifest2->size()) {
```

Comment:
do we need sorting before comparing the sizes ?

Reply:
We don't need it but it is helpful as the comment states.  When things are missing it is easier to see what is missing if both manifests are sorted.

Reply:
so the sorting is done based on the 'file path' length to sort them from short to long path ? just to dump the file names from short to long file names ?

Reply:
No.  They are sorted alphabetically by original path name.  The point I was making is we could of checked that the two manifests had different number of entries first (before the sort) and if they where not the same numbers then just returned false right away.  *But*, the downside of doing that is that it makes looking at the failure harder to see what is missing (or is extra) because the two manifests might have things in different orders.  By taking the time to sort no matter what it makes failures easier to figure out.  Since this is just test code with relatively small numbers the trade off was worth it.

Reply:
ok. yes it would be useful for debugging purpose to see the diff more easily.